api-catalog-ui/frontend/cypress/e2e/detail-page/swagger-rendering.cy.js

```diff
@@ -0,0 +1,148 @@
+/*
+ *
+ *  * This program and the accompanying materials are made available under the terms of the
+ *  * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ *  * https://www.eclipse.org/legal/epl-v20.html
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *
+ *  * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+describe("Swagger rendering", () => {
+
+    beforeEach("Login to API Catalog", () => {
+        cy.login(Cypress.env('username'), Cypress.env('password'));
+    });
+
+    [
+        {
+            "serviceName": "API Gateway",
+            "serviceId": "gateway",
+            "serviceHomepage": `${Cypress.env('catalogHomePage')}`.split("/apicatalog")[0]
+        },
+        {
+            "serviceName": "API Catalog",
+            "serviceId": "apicatalog",
+            "serviceHomepage": `${Cypress.env('catalogHomePage')}`.split("/apicatalog")[0]
+        },
+        {
+            "serviceName": "Mock zOSMF",
+            "serviceId": "mockzosmf",
+            "serviceHomepage": ":10013/"
+        },
+    ].forEach((service) => {
+        it("Rendering Swagger for " + service.serviceName, () => {
+
+            cy.get('#grid-container').contains(service.serviceName).click();
+
+            cy.get('div.tabs-swagger').should('exist');
+
+            cy.get('div.tabs-swagger > div.serviceTab > div.header > h4').should('contain', service.serviceName);
+
+            cy.get('div.tabs-swagger > div.serviceTab > div.header > a:nth-child(2)')
+                .as('serviceHomepage').should("contain", "Service Homepage");
+
+            cy.get('@serviceHomepage').should('have.attr', 'title', 'Open Service Homepage')
+
+            cy.get('div.tabs-swagger')
+                .find('.apiInfo-item')
+                .get('h6:nth-Child(1)')
+                .as('basePath');
+
+            cy.get('@basePath')
+                .get('label')
+                .should('contain', "API Base Path:");
+
+            const regex = new RegExp('^$|\/' + service.serviceId + '\/api(\/v1)?$');
```

Comment:
```suggestion
            const regex = new RegExp(`^\/${service.serviceId}\/api(\/v1)?$`);
```

Reply:
For tested services (API Catalog, API Gateway and Mock zOSMF) the API Base Path is never empty. Therefore removing the disjunction (first alternative is never matched).